src/autodiff_nlsmodel.jl

```diff
@@ -61,3 +73,101 @@ function hprod_residual!(nls :: ADNLSModel, x :: AbstractVector, i :: Int, v ::
   Hiv[:] = ForwardDiff.hessian(x->nls.F(x)[i], x) * v
   return Hiv
 end
+
+function cons(nls :: ADNLSModel, x :: Vector)
+  increment!(nls, :neval_cons)
+  return nls.c(x)
+end
+
+function cons!(nls :: ADNLSModel, x :: Vector, c :: Vector)
+  increment!(nls, :neval_cons)
+  c[1:nls.meta.ncon] = nls.c(x)
+  return c
+end
+
+function jac_coord(nls :: ADNLSModel, x :: Vector)
+  increment!(nls, :neval_jac)
+  J = ForwardDiff.jacobian(nls.c, x)
+  return findnz(J)
```

Comment:
Because `J` is always dense, perhaps we can return something like `([j for i = 1 : n for j = 1 : m], [i for i = 1 : n for j = 1 : m], J[:])` (not tested)? I'm not sure if that would be more or less efficient.

Reply:
J is dense, but may have zeros, and `findnz` will not return those.

Reply:
Ok